### PR TITLE
ci(release): use the new argument

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use pipelineCommit rather than `pipelineVersion` which was  is deprecated

## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [x] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
           pipeline: apm-agent-java-release
-          pipelineVersion: ${{ env.TAG_NAME }}
+          pipelineCommit: ${{ env.TAG_NAME }}
```
Comment (v1v) [inline .github/workflows/release.yml:103]: This caused a regression. Though https://github.com/elastic/apm-pipeline-library/pull/2450 will fix it and make the default behaviour with a higher precedence.

**Review by jackshirazi — APPROVED** [review]

**Comment by v1v** [comment]
@mergifyio update

**Comment by v1v** [comment]
@Mergifyio rebase

**Review by amannocci — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Comment by jackshirazi** [comment]
https://github.com/Mergifyio rebase

**Pull request merged**